src/main/scala/dpla/ingestion3/mappers/providers/NcMapping.scala

```diff
@@ -0,0 +1,190 @@
+      .flatMap(extractStrings)
+      .map(stringOnlyTimeSpan)
+
+    // FIXME This mapping is kind of a mess and should be up for renegotiation with DigitalNC folks
+    // Within <originInfo>, the FIRST <dateCreated keyDate="yes"> and LAST <dateCreated keyDate="yes">.
+//
+//    val rangeDate = dates
+//      .filter(s => s.contains("-") | s.contains("/"))
+//      .lastOption
+//
+//    val nonRangeDates = dates
+//      .filterNot(s => s.contains("-") | s.contains("/"))
+//
+//    val constructedRangeDate = if(nonRangeDates.size > 1) Some(s"${nonRangeDates.head}-${nonRangeDates.last}") else None
+//
+//    val date = (rangeDate, constructedRangeDate, dates.headOption) match {
+//      case (Some(s), _, _) => s // range exists in original data
+//      case (None, Some(s), _) => s // date range was constructed from head and tail values
+//      case (None, None, Some(s)) => s // no range constructed, use first date value
+//      case (_, _, _) => "" // nothing to map
+//    }
+//
+//    Seq(stringOnlyTimeSpan(date))
+  }
+  
+  override def description(data: Document[NodeSeq]): Seq[String] =
+  // <mods:note type='content'>
+    (data \\ "mods" \ "note")
+      .filter(node => filterAttribute(node, "type", "content"))
+      .flatMap(extractStrings)
+
+  override def format(data: Document[NodeSeq]): Seq[String] =
+  // <physicalDescription><form>
+    extractStrings(data \\ "mods" \ "physicalDescription" \ "form")
+
+  override def identifier(data: Document[NodeSeq]): Seq[String] =
+  // <mods:identifier>
+    extractStrings(data \\ "mods" \ "identifier")
+
+  override def language(data: Document[NodeSeq]): Seq[SkosConcept] =
+  // <mods:language><mods:languageTerm>
+    extractStrings(data \\ "mods" \ "language" \ "languageTerm")
+      .map(nameOnlyConcept)
+
+  override def place(data: Document[NodeSeq]): Seq[DplaPlace] =
+  // <mods:subject><mods:geographic>
+    extractStrings(data \\ "mods" \ "subject" \ "geographic")
+      .map(nameOnlyPlace)
+
+  override def publisher(data: Document[NodeSeq]): Seq[EdmAgent] =
+  // <mods:originInfo><mods:publisher>
+    extractStrings(data \\ "mods" \ "originInfo" \ "publisher")
+      .map(nameOnlyAgent)
+
+  override def relation(data: Document[NodeSeq]): ZeroToMany[LiteralOrUri] =
+  // <relatedItem><location><url> and/or <relatedItem><titleInfo><title>
+    (extractStrings(data \\ "mods" \ "relatedItem" \ "location" \ "url") ++
+      extractStrings(data \\ "mods" \ "relatedItem" \ "titleInfo" \ "title"))
+      .map(eitherStringOrUri)
+
+  override def rights(data: Document[NodeSeq]): AtLeastOne[String] =
+  // <accessCondition type="local rights statements">
+    (data \\ "mods" \ "accessCondition")
+      .filter(node => filterAttribute(node, "type", "local rights statements"))
+      .flatMap(extractStrings)
+
+  override def subject(data: Document[NodeSeq]): Seq[SkosConcept] =
+    // <mods:subject><mods:topic>
+    extractStrings(data \\ "mods" \ "subject" \ "topic").map(nameOnlyConcept)
```

Comment:
For consistency, you could put a line break before `.map(nameOnlyConcept)`

Reply:
done